wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -117,9 +117,34 @@ abstract class Wallet
         } else FutureUtil.unit
     } yield ()
 
+  private def checkRootAccount: Future[Unit] = {
```

Comment:
This silently fails for me. The error does not get propogated for some reason.

Reply:
Ok this is very strange, it _is_ failing for me locally, and i finally got it to spit out an error message with this custom code. However I don't understand why the error is propogating back upwards in the call stack. Do you have any idea? Could you try and "trivially fail" (sys.error(s"trivial fail!")) and see if you get the error to propagate? 

```scala
    accountDAO.read(account.coin, account.index).flatMap {
      case Some(account) =>
        logger.info(s"account=$account")
        if (account.xpub != xpub) {
          val errorMsg =
            s"Divergent xpubs for account=$account. Existing database xpub=${account.xpub}, key manager's xpub=$xpub. " +
              s"It is possible we have a different key manager being used than expected, key manager=$keyManager"
          logger.error(errorMsg)
          Future.failed(new RuntimeException(errorMsg))
        } else {
          logger.info(s"Xpubs matched!")
          FutureUtil.unit
        }
      case None =>
        logger.info(s"No accounts found!!")
        val errorMsg = s"Missing root xpub for account $account in database"
        Future.failed(new RuntimeException(errorMsg))
    }
```

Reply:
#1950 should fix this

Reply:
I don't think so, we have a `.failed()` callback registered already don't we at the very end of Main.scala

```
    startFut.failed.foreach { err =>
      logger.error(s"Error on server startup!", err)
    }
```

Reply:
I've tested and it does work. Not sure why the `logger.error` does not appear, maybe because our `AppConfig` has been stopped and we can't log?

Reply:
Either way, discussion should occur on that PR